nips/11.md

```diff
@@ -0,0 +1,61 @@
+NIP-11
+======
+
+Relay Information Document
+---------------------------
+
+`draft` `optional` `author:scsibug`
+
+Relays may provide server metadata to clients to inform them of capabilities, administrative contacts, and various server attributes.  This is made available as a JSON document over HTTP, on the same URI as the relay's websocket.
+
+When a relay receives an HTTP(s) request with an `Accept` header of `application/nostr+json` to a URI supporting WebSocket upgrades, they SHOULD return a document with the following structure.
+
+```json
+["NOSTR_SERVER_INFO",
+{
+  name: <string identifying relay>,
+  descr: <string with detailed information>,
+  pubkey: <administrative contact pubkey>,
+  email: <administrative contact email address>,
+  supported_nips: <a list of NIPS supported by the relay>,
+  software: <string identifying relay software URL>,
+  version: <string version identifier>
+}]
+```
+
+Additional fields may be present, but they MUST be prefixed with `x-` to indicate they are not standard.
+
+Any field may be omitted, and clients MUST ignore any additional fields they do not understand.
+
+Field Descriptions
+-----------------
+
+### Name ###
+
+A relay may select a `name` for use in client software.  This is a string, and SHOULD be less than 30 characters to avoid client truncation.
+
+### Description ###
+
+Detailed plain-text information about the relay may be contained in the `descr` string.  It is recommended that this contain no markup, formatting or line breaks for word wrapping, and simply use double newline characters to separate paragraphs.  There are no limitations on length.
+
+### Pubkey ###
+
+An administrative contact may be listed with a `pubkey`, in the same format as Nostr events (32-byte hex for a `secp256k1` public key).  If a contact is listed, this provides clients with a recommended address to send encrypted direct messages (See `NIP-04`) to a system administrator.  Expected uses of this address are to report abuse or illegal content, file bug reports, or request other technical assistance.
+
+Relay operators have no obligation to respond to direct messages.
+
+### Email ###
+
+An alternative contact may be listed under the `email` field as well, with the same purpose as `pubkey`.
+
```

Comment:
I don't like the email field. I don't like email in general. I don't think we should encourage email usage.
If someone wants to return this they still can -- but email will always be a manual thing anyway, so it doesn't have to be standardized as a field.

Reply:
Email is optional; I think it's just good for anyone who may have an existing admin@ or abuse@ address they want to use to make that available.

Reply:
What if we called this field "contact" instead of "email" and then you could put an email there, but also other things?

Reply:
That sounds good.  if we are going to make it generic like that, I think i'll provide an example or mention that the "mailto:" scheme could be used to indicate an email address.  That leaves it open to other schemes, including just a simple web page/form.